web/src/settings_components.ts

```diff
@@ -49,7 +75,34 @@ export function get_realm_time_limits_in_minutes(property) {
     return val;
 }
 
-export function get_property_value(property_name, for_realm_default_settings, sub, group) {
+type RealmSetting = Omit<typeof realm, "realm_authentication_methods"> & {
+    realm_authentication_methods: Record<string, boolean>;
+};
+type RealmSettingProperties = keyof RealmSetting | "realm_org_join_restrictions";
+
+type RealmUserSettingDefaultType = typeof realm_user_settings_defaults;
+type RealmUserSettingDefaultProperties =
+    | keyof RealmUserSettingDefaultType
+    | "email_notification_batching_period_edit_minutes";
+
+type StreamSettingProperties = keyof StreamSubscription | "stream_privacy" | "is_default_stream";
```

Comment:
There are cases where `property_name` value passed to `get_property_value` is not present in the `realm` or `sub` object.